pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -227,7 +227,11 @@ func (t *ResolvedPipelineRunTask) parentTasksSkip(facts *PipelineRunFacts) bool
 	stateMap := facts.State.ToMap()
 	node := facts.TasksGraph.Nodes[t.PipelineTask.Name]
 	for _, p := range node.Prev {
-		if stateMap[p.Task.HashKey()].Skip(facts) {
+		parentTask := stateMap[p.Task.HashKey()]
+		if parentTask.Skip(facts) {
+			if facts.ScopeWhenExpressionsToTask && parentTask.PipelineTask.WhenExpressions != nil {
```

Comment:
are there any cases we should think about where, parent task is skipped, it HAS a when expression, but the when expression isnt the reason it was skipped?

(im wondering if we should be checking more than just `parentTask.PipelineTask.WhenExpressions != nil`, i.e. checking that they are present AND that they evaluated to skip)